platform/node/src/node_map.cpp

```diff
@@ -682,6 +682,19 @@ void NodeMap::AddImage(const Nan::FunctionCallbackInfo<v8::Value>& info) {
 
     std::unique_ptr<uint8_t[]> data = std::make_unique<uint8_t[]>(imageLength);
     std::copy(imageDataBuffer, imageDataBuffer + imageLength, data.get());
+    
+    auto imageData = data.get();
+    //Pre-multiply the image alpha channel
+    for (size_t i =0 ; i < imageLength ;i+=4) {
+        uint8_t& r = imageData[i + 0];
+        uint8_t& g = imageData[i + 1];
+        uint8_t& b = imageData[i + 2];
+        uint8_t& a = imageData[i + 3];
+        r = (r * a + 127) / 255;
+        g = (g * a + 127) / 255;
+        b = (b * a + 127) / 255;
+    }
```

Comment:
We already have this code in `src/mbgl/util/premultiply.hpp`. Instead, can we move that header to the public include directory, and create an, `UnassociatedImage`, then call `util::premultiply` on it?